lib/widgets/molecules/recommended_mentors_filters.dart

```diff
@@ -0,0 +1,251 @@
+    _languagesController.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final ThemeData theme = Theme.of(context);
+    final AppLocalizations l10n = AppLocalizations.of(context)!;
+
+    return Scaffold(
+      appBar: AppBar(title: Text(l10n.exploreSearchFilterTitle)),
+      body: Padding(
+        padding: const EdgeInsets.all(16),
+        child: Column(
+          children: [
+            Column(children: [
+              _Expertiese(skills: _selectedSkills),
+              _AutocompletePicker(
+                fieldName: "Language",
+                controller: _languagesController,
+                options: ExploreCardFiltersProvider.languages,
+                optionsTranslations: l10n.exploreSearchFilterLanguages,
+                selectedOptions: widget.filtersProvider.selectedLanguages,
+              ),
+              _AutocompletePicker(
+                fieldName: "Countries",
+                controller: _countriesController,
+                options: ExploreCardFiltersProvider.countries,
+                optionsTranslations: l10n.exploreSearchFilterCountries,
+                selectedOptions: widget.filtersProvider.selectedCountries,
+              ),
+            ]),
+            Align(
+              alignment: Alignment.centerLeft,
+              child: TextButton.icon(
+                icon: const Icon(Icons.tune),
+                label: const Text("Advanced filters"),
+                onPressed: null,
+              ),
+            ),
+            Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: [
+                TextButton(
+                  style: TextButton.styleFrom(
+                    minimumSize: Dimensions.bigButtonSize,
+                  ),
+                  child: Text("Clear",
+                      style: theme.textTheme.labelLarge
+                          ?.copyWith(color: theme.colorScheme.secondary)),
+                  onPressed: () {
+                    setState(() {
+                      _countriesController.clearTags();
+                      _languagesController.clearTags();
+                      _selectedSkills.clear();
+                    });
+                  },
+                ),
+                ElevatedButton(
+                  style: ElevatedButton.styleFrom(
+                    minimumSize: Dimensions.bigButtonSize,
+                    backgroundColor: theme.colorScheme.surface,
+                    disabledBackgroundColor: theme.colorScheme.surface,
+                    textStyle: theme.textTheme.labelLarge,
+                  ),
+                  child: const Text("Apply"),
+                  onPressed: () {
+                    widget.filtersProvider.setAll(
+                        _countriesController.getTags?.toSet() ?? {},
+                        _languagesController.getTags?.toSet() ?? {},
+                        _selectedSkills);
+
+                    Navigator.of(context).pop();
```

Comment:
I tested this and every time I press "Apply" I see the following error in the console: `A TextEditingController was used after being disposed.`

Seems like a lifecycle management issue, which can sometimes be very tricky to troubleshoot .